### PR TITLE
e2e: notice FailedScheduling or Preempted events

### PR DESCRIPTION
**Pull request by sjenning**
**What this PR does / why we need it**:

We have had issues with CI clusters being overcommitted in the past leading situations that are difficult to diagnose or reproduce.  This should increase visibility when that happens.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: test/e2e/util/util.go [hunk 1]
```diff
@@ -294,6 +294,22 @@ func EnsureNoCrashingPods(t *testing.T, ctx context.Context, client crclient.Cli
...
+	t.Run("NoticePreemptionOrFailedScheduling", func(t *testing.T) {
+		namespace := manifests.HostedControlPlaneNamespace(hostedCluster.Namespace, hostedCluster.Name).Name
+
+		var eventList corev1.EventList
+		if err := client.List(ctx, &eventList, crclient.InNamespace(namespace)); err != nil {
+			t.Fatalf("failed to list events in namespace %s: %v", namespace, err)
+		}
+		for _, event := range eventList.Items {
+			if event.Reason == "FailedScheduling" || event.Reason == "Preempted" {
+				t.Logf("error: non-fatal, observed FailedScheduling or Preempted event: %s", event.Message)
```
Comment (sjenning) [inline test/e2e/util/util.go:307]: `error: ` is to trigger prow syntax highlight
https://github.com/openshift/release/blob/d61b6d2ee203cdd872ddf75e9b8658836cb13c42/core-services/prow/02_config/_config.yaml#L54

**Review by davidvossel — COMMENTED** [review]
Would it make sense to add this logging to the generic tear down logic?

so, for example, if a test fails early for some reason, the teardown hook could check for preemptive pod events before tearing down the hosted cluster which might help give people a quick hint there was overcommit issues?

File: test/e2e/util/util.go (hunk 1 above)
Comment (enxebre) [reply test/e2e/util/util.go:307]: probably useful to put that in a code comment instead

File: test/e2e/nodepool_upgrade_test.go [hunk 2]
```diff
@@ -110,6 +110,7 @@ func TestUpgradeNodePool(t *testing.T) {
 
 	e2eutil.EnsureNodeCountMatchesNodePoolReplicas(t, ctx, client, guestClient, hostedCluster.Namespace)
 	e2eutil.EnsureNoCrashingPods(t, ctx, client, hostedCluster)
+	e2eutil.NoticePreemptionOrFailedScheduling(t, ctx, client, hostedCluster)
```
Comment (enxebre) [inline test/e2e/nodepool_upgrade_test.go:113]: I think this needs rebase and so it can be added to the two existing tests in this file now.

**Comment by enxebre** [comment]
Agree with @davidvossel we are running
```
	e2eutil.EnsureNodeCountMatchesNodePoolReplicas(t, ctx, client, guestClient, hostedCluster.Namespace)
	e2eutil.EnsureNoCrashingPods(t, ctx, client, hostedCluster)
	e2eutil.EnsureHCPContainersHaveResourceRequests(t, ctx, client, hostedCluster)
	e2eutil.EnsureAllContainersHavePullPolicyIfNotPresent
	e2e.NoticePreemptionOrFailedScheduling
	e2eutil.EnsureNoPodsWithTooHighPriority(t, ctx, client, hostedCluster)
```

pretty much for all tests, it'd be nice to refactor that and remove the need to maintain the list for each test.

**Comment by enxebre** [comment]
https://github.com/openshift/hypershift/pull/1332#issuecomment-1114886765
Created https://github.com/openshift/hypershift/pull/1345

**Pull request merged**